### PR TITLE
CB-14129 Update tag specifications

### PR DESCRIPTION
**Pull request by agov**
Updated AWS, Azure, GCP and OpenStack tag specifications:
- Reflect the latest restrictions (see links in Jira)
- Fixed '-' issue where it was interpreted as a range
- Leading and trailing spaces are restricted

See detailed description in the commit message.

### CODE REVIEW

File: environment/src/test/java/com/sequenceiq/environment/environment/validation/validators/TagValidatorTest.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     public static final String BAD_KEY = "azureprefix";
```
Comment (horadla23) [inline environment/src/test/java/com/sequenceiq/environment/environment/validation/validators/TagValidatorTest.java:36]: I would extend this or I would add more different bad values to cover more cases mentioned in links in the ticket.

Comment (agov) [reply environment/src/test/java/com/sequenceiq/environment/environment/validation/validators/TagValidatorTest.java:36]: I agree, also not only Azure should be tested. But I'm not sure if it should be within the scope of this ticket.

Comment (horadla23) [reply environment/src/test/java/com/sequenceiq/environment/environment/validation/validators/TagValidatorTest.java:36]: I think if we don't extend it here in this PR, no one will later.

Comment (agov) [reply environment/src/test/java/com/sequenceiq/environment/environment/validation/validators/TagValidatorTest.java]: Updated tests as agreed.

Comment (horadla23) [reply environment/src/test/java/com/sequenceiq/environment/environment/validation/validators/TagValidatorTest.java]: Thank you!

**Review by horadla23 — APPROVED** [review]

**Pull request merged**